app/services/anthropic_provider.py

```
@@ -0,0 +1,106 @@
+        """Validate Anthropic API key."""
+        try:
+            client = AsyncAnthropic(api_key=api_key)
+            await client.messages.create(
+                model="claude-3-haiku-20240307",
+                max_tokens=1,
+                messages=[{"role": "user", "content": "test"}]
+            )
+            return True
+        except anthropic.AuthenticationError:
+            return False
+        except Exception:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)